validator/project_validator.go

```diff
@@ -128,11 +125,19 @@ var projectSyntaxValidators = []projectValidator{
 }
 
 // Functions used to validate the semantics of a project configuration file.
-var projectSemanticValidators = []projectValidator{
+var projectWarningValidators = []projectValidator{
 	checkTaskCommands,
 	checkTaskGroups,
 	checkLoggerConfig,
 	checkTaskTimeout,
+	checkProjectFieldsWarnings,
```

Comment:
I think it makes sense to refactor the error checkers to all begin with name "validateXyz", and have warning checkers abide by name "checkXyz" for consistency.  Since functions like checkProjectFieldsWarnings are already in the warning validator I think it's redundant to end the function name with "Warnings".  I think checkProjectFields would suffice.  Same goes for all other functions.